app/scripts/modules/core/src/managed/constraints/constraintRegistry.tsx

```diff
@@ -13,13 +13,21 @@ const throwUnhandledStatusError = (status: string) => {
 
 const { NOT_EVALUATED, PENDING, FAIL, OVERRIDE_PASS, OVERRIDE_FAIL } = StatefulConstraintStatus;
 
+export interface IConstraintOverrideAction {
+  title: string;
+  pass: boolean;
+}
+
 interface IStatefulConstraintConfig {
   iconName: IconNames;
-  shortSummary: (constraint: IStatefulConstraint) => JSX.Element | string;
+  shortSummary: (constraint: IStatefulConstraint) => React.ReactNode;
+  overrideActions: { [status in StatefulConstraintStatus]?: IConstraintOverrideAction[] };
```

Comment:
@vigneshm for this new field I went with the map-of-statuses approach we discussed in a previous PR. I ended up really enjoying it and I think I'll probably go back and switch the `shortSummary` over to that API as well. Thanks for pointing it out!